packages/puppeteer-core/src/bidi/Frame.ts

```diff
@@ -474,8 +475,11 @@ export class BidiFrame extends Frame {
     );
   }
 
-  override waitForDevicePrompt(): never {
-    throw new UnsupportedOperation();
+  override waitForDevicePrompt(
+    options: WaitTimeoutOptions = {},
+  ): Promise<DeviceRequestPrompt> {
+    const {timeout = this.timeoutSettings.timeout(), signal} = options ?? {};
```

Comment:
Already set in the parameters.
```suggestion
    const {timeout = this.timeoutSettings.timeout(), signal} = options;
```